internal/oidc/clientregistry/clientregistry.go

```diff
@@ -11,47 +11,109 @@ import (
 
 	"github.com/coreos/go-oidc/v3/oidc"
 	"github.com/ory/fosite"
+	"k8s.io/apimachinery/pkg/api/errors"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	configv1alpha1 "go.pinniped.dev/generated/latest/apis/supervisor/config/v1alpha1"
+	supervisorclient "go.pinniped.dev/generated/latest/client/supervisor/clientset/versioned/typed/config/v1alpha1"
+	"go.pinniped.dev/internal/oidc/oidcclientvalidator"
+	"go.pinniped.dev/internal/oidcclientsecretstorage"
+	"go.pinniped.dev/internal/plog"
 )
 
-// Client represents a Pinniped OAuth/OIDC client.
+// PinnipedCLIClientID is the client ID of the statically defined public OIDC client which is used by the CLI.
+const PinnipedCLIClientID = "pinniped-cli"
+
+// Client represents a Pinniped OAuth/OIDC client. It can be the static pinniped-cli client
+// or a dynamic client defined by an OIDCClient CR.
 type Client struct {
 	fosite.DefaultOpenIDConnectClient
 }
 
-func (c Client) GetResponseModes() []fosite.ResponseModeType {
-	// For now, all Pinniped clients always support "" (unspecified), "query", and "form_post" response modes.
-	return []fosite.ResponseModeType{fosite.ResponseModeDefault, fosite.ResponseModeQuery, fosite.ResponseModeFormPost}
-}
-
-// It implements both the base, OIDC, and response_mode client interfaces of Fosite.
+// Client implements the base, OIDC, and response_mode client interfaces of Fosite.
 var (
 	_ fosite.Client              = (*Client)(nil)
 	_ fosite.OpenIDConnectClient = (*Client)(nil)
 	_ fosite.ResponseModeClient  = (*Client)(nil)
 )
 
-// StaticClientManager is a fosite.ClientManager with statically-defined clients.
-type StaticClientManager struct{}
+func (c Client) GetResponseModes() []fosite.ResponseModeType {
```

Comment:
```suggestion
func (c *Client) GetResponseModes() []fosite.ResponseModeType {
```

?

Reply:
Updated in 88f611d31a07382565ca78485980023ac8ef2145.